Clarify a bit why server might send retry configs (#490)
tlswg/draft-ietf-tls-esni: draft-ietf-tls-esni.md

: The serialized and encrypted ClientHelloInner structure, encrypted using HPKE
as described in {{real-ech}}.

> When a client offers the `outer` version of an "encrypted_client_hello" extension, 
> then, for example if decryption fails, the server MAY include an "encrypted_client_hello"
> extension in its EncryptedExtensions message with the following payload:

Reviewer: ```suggestion
When a client offers the `outer` version of an "encrypted_client_hello"
extension, the server MAY include an "encrypted_client_hello" extension in its
EncryptedExtensions message, as described in {{client-facing-server}}, with the
following payload:
```

Rather than duplicate text, this just drops a forward pointer to where the rationale is spelled out in more detail.
